main.c がコンパイルエラーで make が失敗する

## 概要
`make` を実行すると `main.c` で5つのコンパイルエラーが発生し、ビルドが失敗します。

## 再現手順
```shell
cd advanced_programming_signal_processing
make
```

## エラー内容
```
main.c:45:14: error: call to undeclared library function 'sqrt' with type 'double (double)'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
main.c:134:9: error: incompatible integer to pointer conversion assigning to 'char *' from 'int' [-Wint-conversion]
main.c:136:9: error: incompatible integer to pointer conversion assigning to 'char *' from 'int' [-Wint-conversion]
main.c:138:9: error: incompatible integer to pointer conversion assigning to 'char *' from 'int' [-Wint-conversion]
main.c:140:9: error: incompatible integer to pointer conversion assigning to 'char *' from 'int' [-Wint-conversion]
```

## 原因
1. `sqrt()` を使用しているが `#include <math.h>` が欠落している
2. `strchr()` の代入文で括弧が不足しており、演算子の優先度により `!=` が `=` より先に評価され、比較結果の `int` が `char *` に代入される
   - `p = strchr(argv[5], 'c') != NULL` → `p = (strchr(argv[5], 'c') != NULL)` と解釈される
   - 正しくは `(p = strchr(argv[5], 'c')) != NULL`

--- a/advanced_programming_signal_processing/main.c
+++ b/advanced_programming_signal_processing/main.c
@@ -3,6 +3,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <limits.h>
+#include <math.h>
 //#include <omp.h>
 
 void templateMatchingGray(Image *src, Image *template, Point *position, double *distance)
@@ -131,13 +132,13 @@ int main(int argc, char **argv)
 	if (argc == 6)
 	{
 		char *p = NULL;
-		if (p = strchr(argv[5], 'c') != NULL)
+		if ((p = strchr(argv[5], 'c')) != NULL)
 			clearResult(output_name_txt);
-		if (p = strchr(argv[5], 'w') != NULL)
+		if ((p = strchr(argv[5], 'w')) != NULL)
 			isWriteImageResult = 1;
-		if (p = strchr(argv[5], 'p') != NULL)
+		if ((p = strchr(argv[5], 'p')) != NULL)
 			isPrintResult = 1;
-		if (p = strchr(argv[5], 'g') != NULL)
+		if ((p = strchr(argv[5], 'g')) != NULL)
 			isGray = 1;
 	}
 

